### PR TITLE
Move FRI remainder module under ext2 in stdlib

### PR DESCRIPTION
**Pull request by bobbinth**
This PR moves FRI module in stdlib from `std::crypto::fri` to `std::crypto::fri::ext2fri` to make the fact that the code assumes that FRI is run in the quadratic extension of the base field. It also makes minor updates to the docs in the `ext2fri` module - but there should be no actual code changes in this PR.

A couple of notes:
1. @Al-Kindi-0: it would be great, at some point, to make a short stand-alone write up about the technique. This way we can reference it from this module.
2. I think the way it currently works, `verify_remainder` procedure checks that `p` is of degree 7 or smaller. But I think we need to check that the degree is exactly 7. This could be easily done by checking that the most-significant coefficient of `p` is non-zero - but I wanted to check if you guys have the same understanding.

### CODE REVIEW

**Comment by Al-Kindi-0** [comment]
Thank you @bobbinth !


> This PR moves FRI module in stdlib from `std::crypto::fri` to `std::crypto::fri::ext2fri` to make the fact that the code assumes that FRI is run in the quadratic extension of the base field. It also makes minor updates to the docs in the `ext2fri` module - but there should be no actual code changes in this PR.
> 
> A couple of notes:
> 
>     1. @Al-Kindi-0: it would be great, at some point, to make a short stand-alone write up about the technique. This way we can reference it from this module.
> 

Yes, sounds like a great idea. I will improve on the note I shared previously to make it clearer so that we can make a blog-post about the technique.

>     2. I think the way it currently works, `verify_remainder` procedure checks that `p` is of degree 7 or smaller. But I think we need to check that the degree is exactly 7. This could be easily done by checking that the most-significant coefficient of `p` is non-zero - but I wanted to check if you guys have the same understanding.
My understanding is that FRI checks that a certain polynomial is of degree less than an upper bound. Thus the remainder is also checked to be less than an (expected) upper bound.
This is in fact what is done in Winterfell, if I am understanding [this](https://github.com/facebook/winterfell/blob/83872ec59147def68c9bd0386d42820ca3c85ba2/fri/src/verifier/mod.rs#L351) portions of the code correctly.

File: stdlib/asm/crypto/fri/ext2fri.masm [hunk 1]
```diff
@@ -1,33 +1,29 @@
-#! Given starting memory address to q i.e. 64 polynomial evaluations ( FRI codeword ), this routine computes
-#! τ, by first non-deterministically interpolating polynomial p ( of degree at max 7 i.e. with 8 coefficients )
-#! using advice provider ( i.e. computation performed outside of the VM ) and then sequentially hashing coefficients
-#! of q, p ( in order ), using Rescue Prime Optimized hash function.
+#! Given starting memory address of a FRI codeword of size 64 of q, this routine computes τ, by
```
Comment (Al-Kindi-0) [inline stdlib/asm/crypto/fri/ext2fri.masm:1]: nit: remove "of" in "of q" ?

**Review by Al-Kindi-0 — APPROVED** [review]
Looks good, thank you!

**Review by itzmeanjan — APPROVED** [review]

**Comment by bobbinth** [comment]
> Yes, sounds like a great idea. I will improve on the note I shared previously to make it clearer so that we can make a blog-post about the technique.

We could even just set up a wiki page in this repo (now that Github allows math expressions).

> My understanding is that FRI checks that a certain polynomial is of degree less than an upper bound. Thus the remainder is also checked to be less than an (expected) upper bound.

Yes, I think that's right. Let's keep it as is.

**Pull request merged**